OracleSOASuite/dockerfiles/12.2.1.4/README.md

```diff
@@ -1,82 +1,186 @@
 
-The containers will be connected using a Docker User Defined network. 
+This documentation provides the steps for the two scenarios to start the containers on the node(s).
 
-## Create a User Defined network
+- **Single node scenario**: all the containers created on single node.
+- **Multinode scenario**: containers can be run on any host connected through Docker swarm.
 
-In this configuration creation of a user defined network will enable the communication between the containers just using container names. For this setup we will use a user defined network using bridge driver.
+> **Note**: In this scenario, the soa-infra page will not display links to `SOA Composer` and `BPM Worklist` but you can access those two applications.
 
-Create a user defined network using the bridge driver by executing the following command :
+To create the Docker network and run containers, follow these steps:
 
-    $ docker network create -d bridge <some name>
-   
-Sample command:
+ 1. [Create a network](#1-create-a-network)
+ 2. [Mount a host directory as a data volume](#2-mount-a-host-directory-as-a-data-volume)
+ 3. [Create the database](#3-create-the-database)
+ 4. [Obtain the SOA 12.2.1.4.0 Docker image](#4-obtain-the-soa-12.2.1.4.0-docker-image)
+ 5. [Create a container for the Administration Server](#5-create-a-container-for-the-administration-server)
+ 6. [Create SOA Managed Server containers](#6-create-soa-managed-server-containers)
+ 7. [Create Oracle Service Bus Managed Server containers](#7-create-oracle-service-bus-managed-server-containers)
+ 8. [Access the Consoles](#8-access-the-consoles)
+ 9. [Clean up the environment](#9-clean-up-the-environment)
+
+### 1. Create a network
+
+#### Single node scenario
+The containers will be connected using a Docker user-defined network.
+
+##### Create a user-defined network
+
+In this configuration, the creation of a user-defined network will enable the communication between the containers just using container names. For this setup we will use a user-defined network using bridge driver.
+
+Create a user-defined network using the bridge driver:
+
+    $ docker network create -d bridge <network name>
+
+For example:
 
     $ docker network create -d bridge SOANet
 
-# Mount a host directory as a data volume
+#### Multinode scenario
+
+The containers on different Docker daemons will be connected using a *Docker overlay network*.
+
+##### Create an overlay network
+
+In this configuration, creation of the Docker overlay network will enable communication between the containers. For this setup we will use a network using *bridge overlay*. In order to connect the containers from different hosts, *Docker swarm* is used. Docker swarm is a container orchestration tool that allows the user to manage multiple containers deployed across multiple node machines.
+
+**Set up Docker swarm**
+
+1.  On any one node, initialize a swarm. This node is called the *swarm manager node*:
+
+        $ docker swarm init
```

Comment:
The command and output should be a single code bloack, i.e.: 

```shell
$ docker swarm init
Swarm initialized: current node (5rsj1c75zpn31cc9lu9yycmi5) is now a manager.
To add a worker to this swarm, run the following command:
docker swarm join --token SWMTKN-1-1r8ap4o4fl5vbt3rtfanh7tr42t8h8lvempi1mmri745bcketd-24av3jhw8etrlf7m1toc8ek9j 100.111.150.225:2377
To add a manager to this swarm, run 'docker swarm join-token manager' and follow the instructions.
```

I also strongly encourage you to use the triple backtick quotes method of creating code blocks instead of using indented lines, because you can specify the type of content for syntax highlighting.